transports/quic/tests/smoke.rs

```diff
@@ -68,614 +31,408 @@ async fn async_std_smoke() {
 
-    let (sync_tx, sync_rx) = oneshot::channel();
+    // Expect the dial to fail since b is not listening on an address.
+    match dial(&mut a_transport, b_send_back_addr).await {
+        Ok(_) => panic!("Expected dial to fail"),
+        Err(error) => {
+            assert_eq!("Handshake with the remote timed out.", error.to_string())
+        }
+    };
 
-    let fut_a = async move {
-        match a.next().await {
-            Some(SwarmEvent::IncomingConnection { .. }) => {}
-            e => panic!("{:?}", e),
-        };
+    let b_addr = start_listening(&mut b_transport, "/ip4/127.0.0.1/udp/0/quic").await;
+    let ((_, a_send_back_addr, _), _) = connect(&mut b_transport, &mut a_transport, b_addr).await;
 
-        match a.next().await {
-            Some(SwarmEvent::ConnectionEstablished { .. }) => {}
-            e => panic!("{:?}", e),
-        };
+    assert_eq!(a_send_back_addr, a_addr);
+}
 
-        match a.next().await {
-            Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
-                message:
-                    RequestResponseMessage::Request {
-                        request: Ping(ping),
-                        channel,
-                        ..
-                    },
-                ..
-            })) => {
-                a.behaviour_mut()
-                    .send_response(channel, Pong(ping))
-                    .unwrap();
-            }
-            e => panic!("{:?}", e),
-        }
+#[cfg(feature = "async-std")]
+#[async_std::test]
+async fn ipv4_dial_ipv6() {
+    let _ = env_logger::try_init();
+    let (a_peer_id, mut a_transport) = create_transport::<quic::async_std::Provider>();
+    let (b_peer_id, mut b_transport) = create_transport::<quic::async_std::Provider>();
 
-        match a.next().await {
-            Some(SwarmEvent::Behaviour(RequestResponseEvent::ResponseSent { .. })) => {}
-            e => panic!("{:?}", e),
-        }
+    let a_addr = start_listening(&mut a_transport, "/ip6/::1/udp/0/quic").await;
+    let ((a_connected, _, _), (b_connected, _)) =
+        connect(&mut a_transport, &mut b_transport, a_addr).await;
 
-        a.behaviour_mut()
-            .send_request(&b_id, Ping(b"another substream".to_vec()));
-
-        assert!(a.next().now_or_never().is_none());
-
-        match a.next().await {
-            Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
-                message:
-                    RequestResponseMessage::Response {
-                        response: Pong(data),
-                        ..
-                    },
-                ..
-            })) => assert_eq!(data, b"another substream".to_vec()),
-            e => panic!("{:?}", e),
-        }
+    assert_eq!(a_connected, b_peer_id);
+    assert_eq!(b_connected, a_peer_id);
+}
 
-        sync_rx.await.unwrap();
+#[cfg(feature = "async-std")]
+#[async_std::test]
+#[ignore] // Transport currently does not validate PeerId. Delete this test?
```

Comment:
Removing this test is fine by me.

Reply:
I changed the comment to say it should be re-written once we do the refactoring.